lib/subscription.js

```diff
@@ -66,6 +79,45 @@ class Subscription extends Entity {
   get isRaw() {
     return this._isRaw;
   }
+
+  /**
+   * Test if the RMW supports content-filtered topics and that this subscription
+   * has been configured with a wellformed content-filter.
+   * @returns {boolean} True if content-filtering will be applied; otherwise false.
+   */
+  isContentFilteringEnabled() {
+    return rclnodejs.isContentFilteringEnabled(this.handle);
+  }
+
+  /**
+   * If the RMW supports content-filtered topics set this subscription's content-filter.
+   * @param {object} contentFilter - The content-filter description.
+   * @param {string} contentFilter.expression - Specifies the criteria to select messages of interest.
+   *  It is similar to the WHERE part of an SQL clause. Clear the current contentFilter if
+   *  the expression is undefined or an empty string.
+   * @param {object[]}  [contentFilter.parameters=undefined] - Array of objects that give values to
+   *  the ‘parameters’ (i.e., "%n" tokens) in the filter_expression. The number of supplied parameters must
+   *  fit with the requested values in the filter_expression (i.e., the number of %n tokens). default: undefined.
+   * @returns {boolean} - True if successful; false otherwise
+   * @see {@link https://www.omg.org/spec/DDS/1.4/PDF|DDS 1.4 specification, Annex B}
+   */
+  setContentFilter(contentFilter) {
+    if (!contentFilter) return false;
+
```

Comment:
re: `isContentFilteringEnabled()`, this will only return true if both:
* the RMW implementation support content-filtering topics (CTF) and
* *the subscription is currently configured with a valid content-filter.

So the use of isContentFilteringEnabled() is limited to identifying if a subscription currently configured with a valid content-filter. I'm not aware of an rcl level api for identifying if an RMW implementation support CTFs.

 I originally implemented without the guard condition like this:
```javascript
  setContentFilter(contentFilter) {
    return contentFilter && contentFilter.expression
      ? rclnodejs.setContentFilter(this.handle, contentFilter)
      : this.clearContentFilter();
  }
```
At the rcl level, passing null or empty string as the content-filter's expression will clear the current content-filter. I propose that we use the implementation above. Thoughts?

Reply:
It sounds that `hasContentFilter()` is more clear.

>At the rcl level, passing null or empty string for conentFilter.expression will clear the current content-filter. I propose that we use the implementation above. Thoughts?

Agree, we can remove `if (!contentFilter) return false;` here, thanks